thewalrus/internal_modes/fock_density_matrices.py

```diff
@@ -178,4 +184,40 @@ def density_matrix_single_mode(cov, pattern, normalize=False, LO_overlap=None, c
     for i in range(K):
         U_K[i::K, i::K] = Uswap
     _, cov = passive_transformation(np.zeros(cov.shape[0]), cov, U_K, hbar=hbar)
-    return _density_matrix_single_mode(cov, N_nums, normalize, LO_overlap, cutoff, hbar)
+
+    # N_nums = list(pattern.values())
+    if method == "recursive":
+        return _density_matrix_single_mode(cov, N_nums, normalize, LO_overlap, cutoff, hbar)
+    cov = project_onto_local_oscillator(cov, M, LO_overlap=LO_overlap, hbar=hbar)
+    num_modes = len(cov) // 2
+    A = Amat(cov)
+    Q = Qmat(cov)
+    fact = 1 / np.sqrt(np.linalg.det(Q).real)
```

Comment:
This is a bit of a confusing name - suggests factorial to me. Also could conflict with the imported `fact` right?

Reply:
Good point